### PR TITLE
DSG-946: Make Notes Editable

### PR DESCRIPTION
**Pull request by jgnuvalence**
## Types of changes

Patch
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [ ] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

File: helm/work-manager/cerbos-policies/transaction_note_resource.yaml [hunk 1]
```diff
@@ -9,6 +9,11 @@ resourcePolicy:
       effect: EFFECT_ALLOW
 
     - actions: ["view"]
+      roles:
+        - transaction-admin
+      effect: EFFECT_ALLOW
+
+    - actions: [ "update" ]
```
Comment (asultan001) [inline helm/work-manager/cerbos-policies/transaction_note_resource.yaml:16]: I think general Agents can still update notes, especially since we audit all changes.

File: service/src/main/java/io/nuvalence/workmanager/service/audit/transaction/TransactionNoteStatusChangedAuditHandler.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * AuditHandler that records an audit event if the status of a Transaction note has changed.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class TransactionNoteStatusChangedAuditHandler implements AuditHandler<TransactionNote> {
```
Comment (asultan001) [inline service/src/main/java/io/nuvalence/workmanager/service/audit/transaction/TransactionNoteStatusChangedAuditHandler.java:23]: Since we are auditing more than just a status change, I think simply calling this class a TransactionNoteChangedAuditHandler may be more appropriate.

File: service/src/main/java/io/nuvalence/workmanager/service/audit/transaction/TransactionNoteStatusChangedAuditHandler.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Override
+    public void publishAuditEvent(OffsetDateTime eventTime, String originatorId) {
+        AuditMapManagementUtility.removeCommonItems(before, after);
+        try {
+            String eventSummary;
+            if (!before.isEmpty() || !after.isEmpty()) {
+                eventSummary =
+                        String.format(
+                                "Transaction note %s changed its dynamic data", transactionNoteId);
```
Comment (asultan001) [inline service/src/main/java/io/nuvalence/workmanager/service/audit/transaction/TransactionNoteStatusChangedAuditHandler.java:52]: Notes don't have dynamic data so maybe reword this to say changed

Comment (jgnuvalence) [reply service/src/main/java/io/nuvalence/workmanager/service/audit/transaction/TransactionNoteStatusChangedAuditHandler.java]: totally true, this was a copy-and-paste error, now it is corrected

File: service/src/test/java/io/nuvalence/workmanager/service/service/NoteServiceTest.java [hunk 4]
```diff
@@ -128,7 +130,7 @@ public void testCreateTransactionNote() {
         String testName = "testName";
         NoteType noteType = new NoteType(UUID.randomUUID(), testName);
 
-        when(noteTypeRepository.findByName(testName)).thenReturn(Optional.of(noteType));
+        when(noteTypeRepository.findByName(eq(testName))).thenReturn(Optional.of(noteType));
```
Comment (Ortega-Dan) [inline service/src/test/java/io/nuvalence/workmanager/service/service/NoteServiceTest.java:133]: this eq here is linted by SonarQube to be not needed. Not a big deal but removing it helps avoid adding a new SQ lint
```suggestion
        when(noteTypeRepository.findByName(testName)).thenReturn(Optional.of(noteType));
```

File: service/src/test/java/io/nuvalence/workmanager/service/audit/TransactionNoteChangedAuditHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                        eq(AuditEventBusinessObject.TRANSACTION),
+                        oldStateArgumentCaptor.capture(),
+                        newStateArgumentCaptor.capture());
+
+        // Extract the captured argument (HashMap)
+        Map<String, String> oldStateProperties = oldStateArgumentCaptor.getValue();
+        Map<String, String> newStateProperties = newStateArgumentCaptor.getValue();
+
+        Assertions.assertTrue(oldStateProperties.size() == 1);
+        Assertions.assertTrue(newStateProperties.size() == 1);
```
Comment (Ortega-Dan) [inline service/src/test/java/io/nuvalence/workmanager/service/audit/TransactionNoteChangedAuditHandlerTest.java:75]: SonarQube recommending Use assertEquals instead. Sometimes not willing to follow those but these changes are trivial and avoid adding SQ lints.

**Review by Ortega-Dan — CHANGES_REQUESTED** [review]
Made some comments with minor changes that still improve codebase and usability. 
Everything looks great to me !

File: swagger.yaml [hunk 6]
```diff
@@ -1230,13 +1263,21 @@ components:
           type: array
           items:
             type: string
+    NoteModelRequest:
```
[the remark this answers is not in the document]
Comment (jgnuvalence) [reply swagger.yaml]: you are right, this was not really part of my code, it already existed but I made the change, good catch, thank you!

File: service/src/test/java/io/nuvalence/workmanager/service/service/NoteServiceTest.java (hunk 4 above)
Comment (jgnuvalence) [reply service/src/test/java/io/nuvalence/workmanager/service/service/NoteServiceTest.java]: thanks Dan, I'll try to keep these things in mind for future PRs, this is done

File: service/src/test/java/io/nuvalence/workmanager/service/audit/TransactionNoteChangedAuditHandlerTest.java (hunk 5 above)
Comment (jgnuvalence) [reply service/src/test/java/io/nuvalence/workmanager/service/audit/TransactionNoteChangedAuditHandlerTest.java]: thanks Dan, I'll try to keep these things in mind for future PRs, this is done

**Review by Ortega-Dan — APPROVED** [review]
:+1:
